sys/kern/vm_map.c

```diff
@@ -191,12 +191,12 @@ void vm_segment_destroy_range(vm_map_t *map, vm_segment_t *seg, vaddr_t start,
 }
 
 void vm_map_delete(vm_map_t *map) {
+  pmap_delete(map->pmap);
```

Comment:
I think you should consider moving this under `map->mtx` lock.

Remember that `vm_page_fault` can repopulate `pmap` with pages as long as there're segments in `vm_map` and nobody holds `map->mtx` lock. This may be a problem when we consider introduction of user-space threads.

Reply:
Hmmm... I'm not sure this is a problem, since any sane implementation of `exit` would wait until there's only one thread left before freeing the memory map, so there shouldn't be any page faults once we get to `vm_map_delete()`.

Reply:
You're right... hopefully that won't be a problem. Nonetheless `pmap` operation order change have made me think of various highly improbable scenarios.